src/validation.cpp

```diff
@@ -1170,6 +1174,21 @@ bool AcceptToMemoryPoolWorker(CTxMemPool& pool, CValidationState& state, const C
         if (!view.HaveInputs(tx))
             return state.Invalid(false, REJECT_DUPLICATE, "bad-txns-inputs-spent");
 
+        // Extract peg-in inputs, pass set to mempool entry. Validity of pegins checked in CheckInputs
+        for (unsigned int i = 0; i < tx.vin.size(); i++) {
+            if (tx.vin[i].m_is_pegin) {
+                // Quick check of pegin witness to extract db entry
+                if (tx.wit.vtxinwit.size() <= i || tx.wit.vtxinwit[i].pegin_witness.stack.size() < 6) {
+                    return state.Invalid(false, REJECT_INVALID, "pegin-no-witness");
+                }
+                std::pair<uint256, COutPoint> pegin = std::make_pair(uint256(tx.wit.vtxinwit[i].pegin_witness.stack[2]), tx.vin[i].prevout);
+                if (view.IsWithdrawSpent(pegin)) {
```

Comment:
Could a maliciously crafted pegin with `tx.wit.vtxinwit[i].pegin_witness.stack[2].IsNull() == true` crash the node by triggering the assert in `IsWithdrawSpent`?

Reply:
Yes, for both the prevout txid and the witness item. Great catch. Need to figure out the best way to handle these.